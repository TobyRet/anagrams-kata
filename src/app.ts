import express from "express"
import { anagrams } from './services/anagrams'

export const app = express();

app.get(/\/(.+)/, async (req, res) => {
  const words = req.path.slice(1)
  const anagramsResponse = await anagrams(words)
  res.contentType('application/json')
  res.send(anagramsResponse);
});