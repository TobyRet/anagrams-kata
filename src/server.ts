import { Application } from 'express'

export const startServer = (app: Application, port: number) => {
  return app.listen(port, () => {
    console.log(`host listening at http://localhost:${port}`);
  });
};