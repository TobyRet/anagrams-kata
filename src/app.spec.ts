import request from 'supertest'
import * as http from 'http'
import { startServer } from './server'
import { app } from './app'

describe('GET /*', () => {
  let server: http.Server;

  beforeAll(() => {
    server = startServer()
  });

  afterAll((done) => {
    server.close(done);
  });

  it('should return 200 and anagrams if they exist', async () => {
    const response = await request(app).get('/crepitus')

    expect(response.status).toBe(200)
    expect(response.headers["content-type"]).toMatch(/json/);
    expect(response.body).toMatchSnapshot()
  })

  it('should return 200 and no anagrams if they do not exist', async () => {
    const response = await request(app).get('/trr')

    expect(response.status).toBe(200)
    expect(response.headers["content-type"]).toMatch(/json/);
    expect(response.body).toEqual([])
  })
})