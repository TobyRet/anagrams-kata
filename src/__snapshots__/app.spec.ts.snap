// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /* should return 200 and anagrams if they exist 1`] = `
[
  "cecities",
  "cercises",
  "cercuses",
  "cereuses",
  "cerrises",
  "cerusite",
  "cesspits",
  "cestuses",
  "circiter",
  "circuits",
  "circuses",
  "cissuses",
  "cistuses",
  "citesses",
  "citruses",
  "creepers",
  "creepier",
  "creepies",
  "creperie",
  "crepiest",
  "crepitus",
  "cressets",
  "cripeses",
  "crispers",
  "crispest",
  "crispier",
  "critters",
  "critturs",
  "cruisers",
  "cruppers",
  "crustier",
  "crusties",
  "cuppiest",
  "cuprites",
  "curettes",
  "curriers",
  "curtsies",
  "cuscuses",
  "cutesier",
  "cutpurse",
  "eccrisis",
  "eccritic",
  "ecesises",
  "epeeists",
  "epicists",
  "epicures",
  "epitrite",
  "erecters",
  "eristics",
  "estrepes",
  "estruses",
  "eucrites",
  "eucritic",
  "eupeptic",
  "eutectic",
  "icterics",
  "iritises",
  "pepperer",
  "peppiest",
  "percepts",
  "peripter",
  "perriers",
  "persists",
  "perspire",
  "perusers",
  "pesterer",
  "pestiest",
  "pettiest",
  "picrites",
  "picritic",
  "pictures",
  "piecrust",
  "piercers",
  "pietists",
  "pipettes",
  "piripiri",
  "pittites",
  "pituites",
  "precepts",
  "precipes",
  "preciser",
  "precises",
  "precures",
  "precurse",
  "preerect",
  "preppier",
  "preppies",
  "preprice",
  "prepuces",
  "pressers",
  "pressies",
  "pressure",
  "presters",
  "preterit",
  "pretests",
  "prettier",
  "pretties",
  "priciest",
  "priseres",
  "prissier",
  "prissies",
  "pruritic",
  "pruritus",
  "puristic",
  "purities",
  "purpures",
  "purpuric",
  "pursiest",
  "pursuers",
  "pursuits",
  "purtiest",
  "pussiest",
  "putterer",
  "puttiers",
  "receipts",
  "recesses",
  "reciters",
  "recruits",
  "rectitic",
  "rectitis",
  "rectress",
  "reerects",
  "reissuer",
  "reissues",
  "reperuse",
  "reprices",
  "reprises",
  "repursue",
  "rescript",
  "rescuers",
  "resecure",
  "resetter",
  "resister",
  "respects",
  "respires",
  "respites",
  "restress",
  "restrict",
  "retirees",
  "retirers",
  "reutters",
  "rictuses",
  "rispetti",
  "ruptures",
  "ruscuses",
  "rustiest",
  "ruttiest",
  "scepters",
  "scissure",
  "scripter",
  "scuppers",
  "scurrier",
  "scurries",
  "scutters",
  "secreter",
  "secretes",
  "securers",
  "securest",
  "seepiest",
  "seisures",
  "sericite",
  "sesterce",
  "sestette",
  "sestetts",
  "sissiest",
  "speccies",
  "specters",
  "speisses",
  "spiciest",
  "spiriest",
  "spiritus",
  "spitters",
  "sprucest",
  "sprucier",
  "spurrers",
  "spurrier",
  "spurries",
  "spurters",
  "sputters",
  "steepers",
  "steepest",
  "steerers",
  "steppers",
  "stipites",
  "stirrers",
  "stirrups",
  "stresses",
  "stricter",
  "stripers",
  "stripier",
  "stripper",
  "strutter",
  "stutters",
  "suitress",
  "superset",
  "suppress",
  "sureties",
  "surprise",
  "surucucu",
  "suspects",
  "suspires",
  "susurrus",
  "tectites",
  "terriers",
  "testiest",
  "tiercets",
  "tippiest",
  "tipsiest",
  "tipsters",
  "titterer",
  "tressier",
  "tressure",
  "trippers",
  "trippets",
  "trippier",
  "trisects",
  "truistic",
  "trussers",
  "trustees",
  "trusters",
  "trustier",
  "trusties",
  "tussises",
  "uppercut",
  "uprisers",
  "upsetter",
  "ureteric",
  "usurpers",
  "uteritis",
  "utterers",
  "utterest",
]
`;
